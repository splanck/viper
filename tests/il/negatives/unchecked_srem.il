il 0.1.2

func @unchecked_srem(%a:i64, %b:i64) -> i64 {
entry(%a:i64, %b:i64):
  %t0 = srem %a, %b
  ret %t0
}
