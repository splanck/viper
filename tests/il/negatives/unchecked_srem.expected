error: unchecked_srem:entry: %2 = srem %t0 %t1: signed remainder must use srem.chk0 (traps on divide-by-zero; matches BASIC MOD semantics)
